cmd/cluster/azure/create.go

```diff
@@ -100,6 +110,22 @@ func (o *RawCreateOptions) Validate(ctx context.Context, opts *core.CreateOption
 	if o.DiskEncryptionSetID != "" && o.ResourceGroupName == "" {
 		return nil, fmt.Errorf("--resource-group-name is required when using --disk-encryption-set-id")
 	}
+
+	// Validate publisher, offer, sku, and version flags are provided if one of them is not empty
+	marketplaceImageInfo := map[string]string{
```

Comment:
You don't need to allocate a map for this, could just

```go
for flag, value := range map[string]*string{
    "marketplace-publisher": &o.MarketplacePublisher,
    /* rest */
} {
    if value == "" {
		return nil, fmt.Errorf("all marketplace flags (i.e. marketplace-publisher, marketplace-offer, marketplace-sku, marketplace-version) are required when using a marketplace image; the following flag was empty: %s", marketplaceImageInfo[flag])
    }
}
```

Reply:
Ok, I updated this a bit more. It's okay if all the flags are empty; if one is set, the others need set as well.